ArdusatSDK.h

```diff
@@ -49,152 +66,532 @@ typedef enum {
+    const char * toJSON(const char * sensorName);
+};
+
+
+/**************************************************************************//**
+ * @class Pressure
+ * @ingroup sensor
+ *
+ * @defgroup pressure
+ * @brief Encapsulates all functionality related to the Pressure Sensor
+ *
+ * This class can be used to initialize, further configure, read, and print
+ * data receieved from the BMP180 on the Adafruit 9 Degrees of Freedom
+ * Board.
+ *
+ * @note This Sensor is not available on the SpaceBoard
+ *
+ * Example Usage:
+ * @code
+ *     Pressure press;                               // Instantiate sensor object
+ *     press.begin();                                // Initialize sensor
+ *     Serial.println(press.readToJSON("pressure")); // Read and print values in JSON
+ * @endcode
+ *****************************************************************************/
+class Pressure: public Sensor {
+  protected:
+    bmp085_mode_t bmp085_mode;
+
+    boolean initialize(void);
+    boolean readSensor(void);
+
+  public:
+    float pressure;
+    Pressure(void);
+    Pressure(bmp085_mode_t mode);
+
+    float altitudeFromSeaLevelPressure(float seaLevelPressure);
+    float seaLevelPressureFromAltitude(float altitude);
+
+    const char * toCSV(const char * sensorName);
+    const char * toJSON(const char * sensorName);
+};
+
+
+/**************************************************************************//**
+ * @class RGBLight
+ * @ingroup sensor
+ *
+ * @defgroup rgblight
+ * @brief Encapsulates all functionality related to the RGBLight Sensor
+ *
+ * This class can be used to initialize, further configure, read, and print
+ * data receieved from the either the TCS34725 board (default) or the ISL29125
+ * board
+ *
+ * @note No RGBLight Sensors are provided with the Space Kit
+ *
+ * Example Usage:
+ * @code
+ *     RGBLight rgb;                          // Instantiate sensor object
+ *     rgb.begin();                           // Initialize sensor
+ *     Serial.println(rgb.readToJSON("rgb")); // Read and print values in JSON
+ * @endcode
+ * Or
+ * @code
+ *     RGBLightTCS rgb;                       // Instantiate sensor object
+ *     rgb.begin();                           // Initialize sensor
+ *     Serial.println(rgb.readToJSON("rgb")); // Read and print values in JSON
+ * @endcode
+ * Or
+ * @code
+ *     RGBLightISL rgb;                       // Instantiate sensor object
+ *     rgb.begin();                           // Initialize sensor
+ *     Serial.println(rgb.readToJSON("rgb")); // Read and print values in JSON
+ * @endcode
+ *****************************************************************************/
+class RGBLight: public Sensor {
+  protected:
+    uint8_t islIntensity;
+    tcs34725IntegrationTime_t tcsIt;
+    tcs34725Gain_t tcsGain;
+
+    boolean initialize(void);
+    boolean readSensor(void);
+    RGBLight(sensor_id_t sensor_id, uint8_t islIntensity, tcs34725IntegrationTime_t tcsIt, tcs34725Gain_t tcsGain);
```

Comment:
So since the RGBLight class defaults to RGBLightTCS, I'm leaving the tcs member variables in the RGBLight class. But since they're configurable only in the constructors of the RGBLightTCS class, there needs to be some constructor in the RGBLight class that I can use when initializing the variables from the RGBLightTCS class. I'm going to leave it protected, but there needs to be at least some constructor like
```
RGBLight(tcs34725IntegrationTime_t tcsIt, tcs34725Gain_t tcsGain);
```
Otherwise, I'd have to set the inherited member variables in the constructor body of RGBLightTCS.